openstack/telemetry/v2/meters/results.go

```diff
@@ -0,0 +1,99 @@
+package meters
+
+import (
+	"reflect"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rackspace/gophercloud"
+)
+
+type MeterListResult struct {
+	MeterId    string `mapstructure:"meter_id"`
+	Name       string `json:"name"`
+	ProjectId  string `mapstructure:"project_id"`
+	ResourceId string `mapstructure:"resource_id"`
+	Source     string `json:"source"`
+	Type       string `json:"type"`
+	Unit       string `json:"user"`
+	UserId     string `mapstructure:"user_id"`
+}
+
+type listResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets any listResult as an array of MeterListResult
+func (r listResult) Extract() ([]MeterListResult, error) {
+	if r.Err != nil {
+		return nil, r.Err
+	}
+
+	var response []MeterListResult
+
+	config := &mapstructure.DecoderConfig{
+		DecodeHook: toMapFromString,
+		Result:     &response,
+	}
+	decoder, err := mapstructure.NewDecoder(config)
+	if err != nil {
+		return nil, err
+	}
+
+	err = decoder.Decode(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+type MeterStatisticsResult struct {
```

Comment:
Same as above: maybe just `Statistics`. Without context, it's vague, but in context it would be called like `meters.Statistics`, which makes perfect sense.